src/inventory_ui.cpp

```diff
@@ -971,10 +972,9 @@ void inventory_column::draw( const catacurses::window &win, const point &p )
 
         const bool selected = active && is_selected( entry );
 
-        entry.drawn_info.text_x_start = x1;
```

Comment:
I just encountered a bug where if you scroll to the second page of the inventory menu and click an entry, the entry at the same place in the first page is selected instead. I think it happens because the drawn info of the entries of the first page are not reset when drawing the second page.

Reply:
Maybe it will be better to use a static vector of pair<rectangle, entry> and update every draw?

Reply:
ok, it is fixed now

Reply:
Seems you only fixed it for the inventory UI. What about the pickup menu and uilist? They seem to be using the same logic so they probably also need fixing.

Reply:
they don't have problem, because pickup list was done in this type of logic as it dont have a entry struct or class. For uilist, I had that kind of problem, and I fixed.